### PR TITLE
CRM-15829 and CRM-15881 for 4.6

### PR DESCRIPTION
**Pull request by ghost**
## CRM-15829 (Relationships and inherited pending memberships)

Add pending memberships to new members of an organisation that has pending memberships
- CRM/Contact/BAO/Relationship -> line ~1458
- CRM/Contact/BAO/Relationship -> line ~1479
- CRM/Contact/BAO/Relationship -> line ~1483

Ensure these new pending memberships for an organisation's employees are not set to active

```
CRM/Member/BAO/Membership -> line 265
```
## CRM-15881 (Contact relationships (employee of))

Clear current employee when deleting a relationship that is the employer
- CRM/Contact/BAO/Relationship -> line ~623

Reload the tab ajax views to reflect removed employer display when deleting a relationship that is the employer
- CRM/Contact/Form/Relationship -> line ~399

If disabling an employer relationship (via the popup modal), unset the is_current_employer, please see comments in code
- CRM/Contact/Form/Relationship -> line ~444

### CODE REVIEW

**Comment by marcov4lente** [comment]
Good Afternoon All,

I'm not sure that this test failure (CRM_Core_BAO_ActionScheduleTest.testContactModifiedAnniversary) relates to the changes that this pull request contains. I have gone through the test's code as well as noticed several other Jenkins test failures of this nature right before this test.

Could anyone perhaps confirm this?

**Comment by colemanw** [comment]
jenkins, retest this

**Comment by colemanw** [comment]
jenkins, retest this please

**Comment by davecivicrm** [comment]
Marco - definitely not related. That test failure is likely related to date boundaries and occurs periodically.

File: CRM/Contact/BAO/Relationship.php [hunk 1]
```diff
@@ -620,7 +620,10 @@ public static function clearCurrentEmployer($id, $action) {
         $sharedContact->id = $relationship->contact_id_a;
         $sharedContact->find(TRUE);
 
-        if ($relationship->relationship_type_id == 4 && $relationship->contact_id_b == $sharedContact->employer_id) {
+        // CRM-15881 UPDATES
+        // changed FROM "...relationship->relationship_type_id == 4..." TO "...relationship->relationship_type_id == 5..."
+        // As the system should be looking for type "employer of" (id 5) and not "sibling of" (id 4)
+        if ($relationship->relationship_type_id == 5 && $relationship->contact_id_b == $sharedContact->employer_id) {
```
Comment (davecivicrm) [inline CRM/Contact/BAO/Relationship.php]: This bug is caused by the fact that relationship_type.id values may be different for different sites. Let's not use the constant '5', but instead get the Employee/Employer relationship type using the name column (which is not translated / edited):
$empRelTypeId = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_RelationshipType', 'Employee of', 'id', 'name_a_b');

File: CRM/Contact/BAO/Relationship.php [hunk 2]
```diff
@@ -1452,6 +1455,20 @@ public static function relatedMemberships($contactId, &$params, $ids, $action =
       }
     }
 
+    // CRM-15829 UPDATES
+    // If we're looking for active memberships we must consider pending (id: 5) ones too.
+    // Hence we can't just call CRM_Member_BAO_Membership::getValues below with the active flag, is it would completely miss pending relatioships.
+
+    $query = 'SELECT * FROM `civicrm_membership_status`';
+    if ($active) {
+      $query .= 'WHERE `is_current_member` = 1 OR `id` = 5';
```
Comment (davecivicrm) [inline CRM/Contact/BAO/Relationship.php]: Please don't use constant '5' for 'Pending' as membership_status.id values are not the same for all sites.
`$pendingStatusId = array_search('Pending', CRM_Member_PseudoConstant::membershipStatus());`

**Comment by marcov4lente** [comment]
Hi @davecivicrm thank you for the valuable feedback. I will make the recommended changes and commit them up!

**Comment by colemanw** [comment]
FYI the newer way to lookup an option id is with 
CRM_Core_Pseudoconstant:: getKey

**Comment by colemanw** [comment]
How is this PR different from #5939? Note that changes in 4.6 automatically get merged-forward into master.

**Comment by marcov4lente** [comment]
Hi @colemanw , there is no difference. Thank you for all your assistance.

**Pull request merged**